packages/hash/frontend/src/components/hooks/useUser.ts

```diff
@@ -45,5 +71,10 @@ export const useUser = (options?: Omit<QueryHookOptions, "errorPolicy">) => {
   //   client.writeQuery();
   // };
 
-  return { user, refetch, loading };
+  return {
+    user,
+    kratosSession,
+    refetch: () => Promise.all([refetchUser, fetchKratosIdentity]),
```

Comment:
It's not really that we need to refetch the session, it's so that we can programmatically "refresh" the values for `user` and `kratosSession` (if the user has logged in/out for example).

We may not need it now as on login we redirect the user to another page, but if we were to implement logging in the user in a modal for instance where there isn't a redirect I think we would need to use this `refetch` method to get the values for `user` and `kratosSession` here.

If we have a logout button that doesn't redirect the user to another page, I think we'd also need to call `refetch` to clear the values for `user` and `kratosSession`.

Reply:
I'm less clear on clearing `kratosSession` but on logout we need to clear the entire ApolloClient cache (including the `user`). We can see how it is when we incorporate it properly with the wider app

Reply:
To be clear, I'm not entirely sure whether having `kratosSession` accessible in react is going to be that useful - it just was for debugging/developing the kratos flows. Perhaps we'll get rid of it, in which case refreshing the react state won't be important.